app/assets/javascripts/components/ReportFilter.jsx

```diff
@@ -116,47 +123,43 @@ class ReportFilter extends React.Component {
             { this.all_categories.map((category, i) => {
               return (
                 <p key={i}>
-                  <input type="checkbox" className="filled-in cat-filter" id={category.name} value={category.name} onClick={this.applyExcludedCategories} defaultChecked={this.props.report_page_params.excluded_categories.indexOf(category.name) < 0} />
+                  <input type="checkbox" className="filled-in cat-filter" id={category.name} value={category.name} onClick={this.applyExcludedCategories} onChange={(e) => {}} checked={this.state.excluded_categories.indexOf(category.name) < 0} />
                   <label htmlFor={ category.name }>{ category.name }</label>
                 </p>
               )
             })}
             { this.all_categories.length < 1 ? <p>None found</p> : '' }
           </div>
         </div>
-      );
-    }
-    genus_search = '';
-    if (this.props.report_page_params.disable_filters == 0) {
-      genus_search = (
-        <div className="filter-controls">
-          <div className="row">
-            <div className="input-field col s12 genus-search-row">
-              <div className='genus-name-label'>GENUS SEARCH</div>
-              <div className="filter-values genus-autocomplete-container">
-                <ReactAutocomplete
-                  inputProps={{ placeholder: 'Search for a genus...' }}
-                  items={this.state.genus_search_items}
-                  shouldItemRender={(item, value) => item == 'None' || item.toLowerCase().indexOf(value.toLowerCase()) > -1}
-                  getItemValue={item => item}
-                  renderItem={(item, highlighted) =>
-                    <div
-                      key={item}
-                      style={{ backgroundColor: highlighted ? '#eee' : 'transparent'}}
-                    >
-                      {item}
-                    </div>
-                  }
-                  value={this.state.genusSearchValue}
-                  onChange={this.handleGenusSearch}
-                  onSelect={this.applyGenusFilter}
-                />
-              </div>
+    );
+    genus_search = (
+      <div className="filter-controls">
+        <div className="row">
+          <div className="input-field col s12 genus-search-row">
+            <div className='genus-name-label'>SEARCH</div>
+            <div className="filter-values genus-autocomplete-container">
+              <ReactAutocomplete
```

Comment:
The slowness here is from (70%) creating 10k+ react objects and (30%) inserting them into the page.  

You don't need to be rendering that many autocomplete suggestions as no one is going to be scrolling through that many.  The simplest way of improving this would to be limiting the number of suggestions by overriding the renderMenu function.  Something like this should work:
```
renderMenu={(items, value, style) =>
   <div style={{ ...style, ...this.menuStyle }} children={items.slice(0, 500)}/> }
```